src/extension.ts

```diff
@@ -180,15 +180,30 @@ async function checkValidAnalyzedProjectDirectory(context: vscode.ExtensionConte
     return true;
 }
 
-// Converts the directory to analyze to an array of directories, if necessary.
+// Converts the directory to analyze to an array of directories if necessary,
+// and replace ${workspaceFolder} with the root workspace vscode directory.
 function normalizeDirsToAnalyze(conf: string|string[]|undefined): string[] {
     if (!conf) {
         return [];
     }
-    if (conf instanceof Array) {
+
+    if (!(conf instanceof Array)) {
+        conf = [conf];
+    }
+
+    if (!vscode.workspace.workspaceFolders) {
         return conf;
     }
-    return [conf];
+
+    if (typeof vscode.workspace.workspaceFolders[0].uri.fsPath !== 'string') {
+        return conf;
+    }
+
+    for (let i = 0; i < conf.length; i++) {
```

Comment:
Would prefer https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/map to be easier to test (not modify original input) but otherwise seems reasonable.

Now that vs code supports marking folders as trusted outside of extensions there's less of a need to manually list workplaces to trust.